### PR TITLE
New RPC Methods

### PR DESCRIPTION
**Pull request by giovannivignone**
# Overview
Implementation of the `eth_getBlockTransactionCountByHash` and `eth_getBlockTransactionCountByNumber` RPC methods according to #139 

## Changes
- Additional methods in node.rs and rpc.rs in the client
- Abstraction of fetching payload logic into a function called get_payload_by_hash in node.rs
- Change from infura -> alchemy in example as recommended RPC url

## ToDo
- [x] Adjust rpc.md 
- [x] Add methods to the client.rs file

### CODE REVIEW

File: client/src/client.rs [hunk 1]
```diff
@@ -336,8 +336,26 @@ impl<DB: Database> Client<DB> {
...
-        self.node.read().await.get_nonce(address, block).await
+    pub async fn get_transaction_count(&self, address: &Address, block: BlockTag) -> Result<u64> {
+        self.node
+            .read()
+            .await
+            .get_transaction_count(address, block)
+            .await
+    }
+
+    pub async fn get_block_transaction_count_by_hash(&self, hash: &Vec<u8>) -> Result<U256> {
```
Comment (ncitron) [inline client/src/client.rs:347]: Can we have both of these functions just return `u64`. They are easier to work with, and it is impossible to have a block fill enough transactions to overflow this. Same goes for these functions in `Node`.

**Review by ncitron — CHANGES_REQUESTED** [review]
Looks pretty good. Just have a couple of minor changes.

Also don't worry about the benchmark ci failing. This is because you are opening from a fork, which doesn't have access to the RPC url secret in this repo. I'll think of a way to handle this in the future.

File: client/src/client.rs [hunk 2]
```diff
@@ -336,8 +336,26 @@ impl<DB: Database> Client<DB> {
         self.node.read().await.get_balance(address, block).await
     }
 
-    pub async fn get_nonce(&self, address: &Address, block: BlockTag) -> Result<u64> {
-        self.node.read().await.get_nonce(address, block).await
+    pub async fn get_transaction_count(&self, address: &Address, block: BlockTag) -> Result<u64> {
```
[the remark this answers is not in the document]
Comment (giovannivignone) [reply client/src/client.rs:339]: This makes total sense. I was hoping you would comment on this as I wasn't sure it if was a good idea to introduce the change. Thank you will remove

**Review by ncitron — APPROVED** [review]
LGTM

**Pull request merged**